### ISSUE TITLE
Data analysis on front-end

### ISSUE THREAD

**Issue by Razvan00Rusu** [SOMAS2022/SOMAS2022#50]
Was wondering if we can get some insight from the maths about what kind of analysis of data we might want to do on the front-end.

At the moment, what exactly is being logged isn't properly defined but the server is currently storing everything in addition to the git commit it is running on and configuration details.

**Comment by Issa-Bqain** [comment]
Log all data relating to input/output for each agent. 

Given an agent, who has X health, etc , they chose to fight. The result was that their new health is Z etc.

This should open up the scope for RL/ML implementations as well as trust/utility rankings.

happy to help with the data analysis once the data logging is complete.

**Comment by alexpondaven** [comment]
Hi, not sure about current progress on this, but just had some ideas about end of game statistics.
- General summary of game
    - How HP pool changes over time
    - How quickly they defeated each boss
- Agent statistics can display the highest performing agents
    - Most valuable player
    - Agent with the highest of a particular attribute at the end/overall (e.g. Damage/trades/health gained)
    - Institutional power
        - Leaders elected throughout the game
        - What strategies or allocations they did
 
Just in general would be cool to have visualisations of how attributes of agents change over time (and would be good to have compatibility with any team-specific agent attributes like bravery). Visualising the social network could also be interesting e.g. How an attribute like trust or our perception of other agents develops over time (but this is probably way too advanced in the given time).
 
It would also be good to have a way to store or visualise information between games. I imagine we'd do some kind of parameter search, and then we could choose a particular metric like utility to plot for different parameters. Not sure how flexible this can be in a UI, but as long as we have a way to store the metrics for each game together with the commit and parameters, it can be plotted somehow.

**Comment by Razvan00Rusu** [comment]
Mostly done through #121. For anything else just open a new issue.
